core/src/server.rs

```diff
@@ -45,3 +55,20 @@ pub async fn run_any_port(config: Config, allocated_port: oneshot::Sender<u16>)
 pub async fn run_on_port(config: Config, port: u16) {
     run(config, Some(port), None).await
 }
+
+async fn show_config(Extension(config): Extension<Config>) -> impl IntoResponse {
+    return config.serialize_pretty()
+}
+
+async fn show_version() -> impl IntoResponse {
+    let mut version = Version::new(0, 0, 0);
+    version = Version::parse(env!("CARGO_PKG_VERSION")).unwrap_or(version);
+
+    let json = serde_json::json!({
+        "MAJOR": version.major,
+        "MINOR": version.minor,
+        "PATCH": version.patch
+    });
+
+    return serde_json::to_string_pretty(&json).unwrap()
```

Comment:
This works, but I had hoped for something in semver that would let me serialize the Version struct to a JSON string directly, but I wasn't quite able to do that (or did I miss something?)

Reply:
With the `serde` feature, the `Version` struct is directly serializable: https://github.com/dtolnay/semver/blob/master/src/serde.rs#L6
So you should be able to pass `&version` directly instead of `&json`.